modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/Oauth2JWKSEndpointTestCase.java

```diff
@@ -72,4 +106,27 @@ private HttpResponse sendGetRequest(HttpClient client, String jwksEndpoint) thro
         HttpResponse response = client.execute(getRequest);
         return response;
     }
+
+    private String getX509CertSHA256Thumbprint(String cert) {
+
+        try {
+            X509Certificate certificate = parseCertificate(cert);
+            JWK parsedJWK = JWK.parse(certificate);
+            return parsedJWK.getX509CertSHA256Thumbprint().toString();
+        } catch (CertificateException | JOSEException e) {
+            Assert.fail("Failed to compute SHA 256 thumb print.", e);
+        }
+        return null;
+    }
+
+    private X509Certificate parseCertificate(String content) throws CertificateException {
+
+        byte[] decodedContent = java.util.Base64.getDecoder().decode(StringUtils.trim(content
+                .replaceAll(BEGIN_CERT, StringUtils.EMPTY)
+                .replaceAll(END_CERT, StringUtils.EMPTY)
```

Comment:
Do we need to replace these parts here since we don't get the begin and end lines in the x5c attribute of the jwks response ?